bread/layout/components/search_select.py

```diff
@@ -0,0 +1,109 @@
+import htmlgenerator as hg
+from django.utils.translation import gettext_lazy as _
+
+import bread.utils
+
+from .icon import Icon
+from .loading import Loading
+from .tag import Tag
+
+
+class SearchSelect(hg.BaseElement):
+    def __init__(
+        self,
+        search_view,
+        query_urlparameter="q",
+        size="xl",
+        searchinput_widgetattributes=None,
+        widgetattributes=None,
+        **elementattributes,
+    ):
+        current_selection_id = widgetattributes["value"][0]
+        # This works inside a formset. Might need to be changed for other usages.
+        current_selection = (
+            getattr(
+                elementattributes["boundfield"].form.instance,
+                elementattributes["fieldname"],
+            )
+            if current_selection_id
+            else None
+        )
```

Comment:
I think this code would be better of with Lazy elements, refering to my big "form elements refactoring" which never happens ;).
When we depend on implementation details from boundfield we can not use the component outside of a Django form. But yeah, it is a mess currently and also not a big issue for now. But it would help in general to decouple the components a bit in the future.